diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 03553029..2b2600c8 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -20,27 +20,5 @@ execute_process(COMMAND cmake -E remove_directory ${CMAKE_BINARY_DIR}/test_resul
 execute_process(COMMAND cmake -E make_directory ${CMAKE_BINARY_DIR}/test_results)
 include_directories(${GTEST_INCLUDE_DIRS})
 
-#============================================================================
-# Do a fake install of ign-math in order to test the examples.
-#============================================================================
-# install to FAKE_INSTALL_PREFIX defined in root CMakeLists.txt
-
-file(MAKE_DIRECTORY ${FAKE_INSTALL_PREFIX})
-
-include(ExternalProject)
-ExternalProject_Add(
-  FAKE_INSTALL
-
-  SOURCE_DIR "${CMAKE_SOURCE_DIR}"
-  EXCLUDE_FROM_ALL 1
-  LOG_CONFIGURE 1
-  LOG_BUILD 1
-  LOG_INSTALL 1
-  CMAKE_ARGS
-    "-DBUILD_TESTING=OFF"
-    "-DCMAKE_INSTALL_PREFIX=${FAKE_INSTALL_PREFIX}"
-)
-
-add_subdirectory(integration)
 add_subdirectory(performance)
 add_subdirectory(regression)

diff --git a/src/python_pybind11/CMakeLists.txt b/src/python_pybind11/CMakeLists.txt
index 22adef2d..4e828b67 100644
--- a/src/python_pybind11/CMakeLists.txt
+++ b/src/python_pybind11/CMakeLists.txt
@@ -145,11 +145,6 @@ if (BUILD_TESTING)
     add_test(NAME ${test}.py COMMAND
       "${PYTHON_EXECUTABLE}" "${CMAKE_SOURCE_DIR}/src/python_pybind11/test/${test}.py")
 
-    set(_env_vars)
-    list(APPEND _env_vars "PYTHONPATH=${FAKE_INSTALL_PREFIX}/lib/python/")
-    list(APPEND _env_vars "LD_LIBRARY_PATH=${FAKE_INSTALL_PREFIX}/lib:$ENV{LD_LIBRARY_PATH}")
-    set_tests_properties(${test}.py PROPERTIES
-      ENVIRONMENT "${_env_vars}")
   endforeach()
 
 endif()
 
